src/main/java/com/angelbroking/smartapi/http/SmartAPIRequestHandler.java

```diff
@@ -1,420 +1,379 @@
+     * @throws JSONException     if there is an error while processing the JSON response from the SmartAPI service
+     */
+
+    public JSONObject getRequest(String apiKey, String url, String accessToken) throws IOException, SmartAPIException, JSONException {
+        Request request = createGetRequest(apiKey, url, accessToken);
+        Response response = client.newCall(request).execute();
+        String body = response.body().string();
+        return new SmartAPIResponseHandler().handle(response, body);
+    }
+
+    /**
+     * Creates a GET request.
+     *
+     * @param url         is the endpoint to which request has to be done.
+     * @param apiKey      is the api key of the Smart API Connect app.
+     * @param accessToken is the access token obtained after successful login
+     *                    process.
+     * @throws IOException
+     */
+    public Request createGetRequest(String apiKey, String url, String accessToken) {
+
+        HttpUrl.Builder httpBuilder = HttpUrl.parse(url).newBuilder();
+
+        String privateKey = apiKey;
+
+        return new Request.Builder().url(httpBuilder.build()).header(Constants.USER_AGENT, USER_AGENT).header(Constants.AUTHORIZATION, "Bearer " + accessToken).header(Constants.CONTENT_TYPE, Constants.APPLICATION_JSON).header(Constants.CLIENT_LOCAL_IP, apiheader.getString(Constants.HEADER_CLIENT_LOCAL_IP)).header(Constants.CLIENT_PUBLIC_IP, apiheader.getString(Constants.HEADER_CLIENT_PUBLIC_IP)).header(Constants.X_MAC_ADDRESS, apiheader.getString(Constants.MAC_ADDRESS)).header(Constants.ACCEPT, apiheader.getString(Constants.ACCEPT)).header(Constants.PRIVATE_KEY, privateKey).header(Constants.X_USER_TYPE, apiheader.getString(Constants.USER_TYPE)).header(Constants.X_SOURCE_ID, apiheader.getString(Constants.SOURCE_ID)).build();
+    }
+
+    /**
+     * Creates a GET request.
+     *
+     * @param url         is the endpoint to which request has to be done.
+     * @param apiKey      is the api key of the Smart API Connect app.
+     * @param accessToken is the access token obtained after successful login
+     *                    process.
+     * @param commonKey   is the key that has to be sent in query param for quote
+     *                    calls.
+     * @param values      is the values that has to be sent in query param like 265,
+     *                    256265, NSE:INFY.
+     */
+    public Request createGetRequest(String url, String commonKey, String[] values, String apiKey, String accessToken) {
+        HttpUrl.Builder httpBuilder = HttpUrl.parse(url).newBuilder();
+        for (int i = 0; i < values.length; i++) {
+            httpBuilder.addQueryParameter(commonKey, values[i]);
+        }
+        return new Request.Builder().url(httpBuilder.build()).header(Constants.USER_AGENT, USER_AGENT).header(Constants.SMART_API_VERSION, "3").header(Constants.AUTHORIZATION, Constants.TOKEN + apiKey + ":" + accessToken).build();

```

Comment:
Use String builder for string concatenation wherever possible.

Reply:
string builder is used for Authorization